components/VersionSection.tsx

```diff
@@ -2,19 +2,39 @@
 
 import { CheckCircle2, Info, Loader2, XCircle } from 'lucide-react';
 import { use, useEffect, useState } from 'react';
+import Markdown from 'react-markdown';
 import { z } from 'zod';
+import Link from '~/components/Link';
+import { Alert, AlertDescription, AlertTitle } from '~/components/ui/Alert';
 import { env } from '~/env';
 import { type getInstallationId } from '~/queries/appSettings';
 import { ensureError } from '~/utils/ensureError';
 import SettingsSection from './layout/SettingsSection';
 import Paragraph from './ui/typography/Paragraph';
 
+type SemVerUpdateType = 'major' | 'minor' | 'patch';
+
 const GithubApiResponseSchema = z.object({
-  status: z.string(),
-  ahead_by: z.number(),
-  behind_by: z.number(),
+  tag_name: z.string().transform((tag) => tag.replace('v', '')), // remove the v to compare with APP_VERSION
+  body: z.string(),
 });
 
+function getSemverUpdateType(
+  currentVersion: string,
+  mainVersion: string,
+): SemVerUpdateType | null {
+  if (currentVersion === mainVersion) return null;
+
+  const [currentMajor, currentMinor, currentPatch] = currentVersion.split('.');
+  const [mainMajor, mainMinor, mainPatch] = mainVersion.split('.');
+
+  if (mainMajor !== currentMajor) return 'major';
+  if (mainMinor !== currentMinor) return 'minor';
+  if (mainPatch !== currentPatch) return 'patch';
```

Comment:
You're comparing these as strings, and only checking if they are different. It is a fairly reasonable assumption, but versions are actually numbers. If you do "1" > "2", it will actually work, but it isn't intentional. 

I think a better way would be to use the zod schema to cast these as numbers, and then to do `currentVersion < newVersion`.